prov/usnic/src/usdf_cq.c

```diff
@@ -324,19 +350,16 @@ usdf_progress_hard_cq(struct usdf_cq_hard *hcq, enum fi_cq_format format)
 			entry = cq->c.soft.cq_head;
 			switch (format) {
 			case FI_CQ_FORMAT_CONTEXT:
-				entry_size = sizeof(*ctx_entry);
 				ctx_entry = (struct fi_cq_entry *)entry;
```

Comment:
This is wrong; the `entry` is always of type `usdf_cq_soft_entry` now, right?  So this whole `switch` block should be replaced with an unconditional filling in of `entry`'s fields and the various `fi_cq_${TYPE}_entry` local variables should just go away.